tests/AblyBroadcasterTest.php

```diff
@@ -325,4 +327,34 @@ protected function getMockRequestWithoutUserForChannel($channel, $token)
 
         return $request;
     }
+
+    public function testLaravelAblyAgentHeader()
+    {
+        $ably = (new AblyFactory())->make([
+            'key' => 'abcd:efgh',
+            'httpClass' => 'Ably\LaravelBroadcaster\Tests\HttpMock',
+        ]);
+        $ably->time();
+        $expectedLaravelHeader = 'ably-php/'.\Ably\Defaults::LIB_VERSION.' '.'php/'.Miscellaneous::getNumeric(phpversion()).' laravel-broadcaster/'. AblyBroadcaster::LIB_VERSION;
+        $this->assertcontains( 'Ably-Agent: '.$expectedLaravelHeader, $ably->http->lastHeaders, 'Expected Laravel broadcaster header in HTTP request' );
+    }
 }
+
+
+class HttpMock extends Http
+{
+    public $lastUrl;
+    public $lastHeaders;
+
+    public function request($method, $url, $headers = array(), $params = array())
+    {
+        $this->lastUrl = $url;
+        $this->lastHeaders = $headers;
+
+        // mock response to /time
+        return array(
+            'headers' => "HTTP/1.1 200 OK\n",
```

Comment:
Clearly this test is passing but I'm a bit confused by this line, the field is called 'headers' but it contains an HTTP start-line? And is there not a neater way to do this than to write raw HTTP response text?

Reply:
I think this is how we have done it in `ably-php` and `ably-php-laravel`. This is just a copy pasted mock from there ... maybe this is the only way to do it.

Reply:
OK that's fair, I don't particularly like it but I won't block this PR on it 🙂